matchbook-sdk-core/src/main/java/com/matchbook/sdk/core/ClientRest.java

```diff
@@ -0,0 +1,52 @@
+package com.matchbook.sdk.core;
+
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+
+import com.matchbook.sdk.core.configs.ClientConnectionManager;
+import com.matchbook.sdk.core.services.Auth;
+import com.matchbook.sdk.core.services.AuthRest;
+import com.matchbook.sdk.core.services.ClientService;
+
+public class ClientRest implements Client {
```

Comment:
We already have a `RestClient` interface in the `clients.rest` subpackage. I fear this can raise up some confusion. I suggest to either rename this class or the other interface (and its implemented classes).

Reply:
i don't like those names too, but let's leave it for now until we find a better name